File: src/test/java/com/ject/studytrip/mission/helper/MissionTestHelper.java
package com.ject.studytrip.mission.helper;

import com.ject.studytrip.mission.domain.model.Mission;
import com.ject.studytrip.mission.domain.repository.MissionRepository;
import com.ject.studytrip.mission.fixture.MissionFixture;
import com.ject.studytrip.stamp.domain.model.Stamp;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class MissionTestHelper {
    @Autowired private MissionRepository missionRepository;

    public Mission saveMission(Stamp stamp) {
        Mission mission = MissionFixture.createMission(stamp);

        return missionRepository.save(mission);
    }

    public Mission saveDeletedMission(Stamp stamp) {
        Mission mission = MissionFixture.createMission(stamp);
        mission.updateDeletedAt();

        return missionRepository.save(mission);
    }

    public Mission saveCompletedMission(Stamp stamp) {
        Mission mission = MissionFixture.createMission(stamp);
        mission.updateCompleted();

        return missionRepository.save(mission);
    }
}

File: src/test/kotlin/com/ject/studytrip/mission/helper/MissionTestHelper.kt
package com.ject.studytrip.mission.helper

import com.ject.studytrip.mission.domain.model.Mission
import com.ject.studytrip.mission.domain.repository.MissionRepository
import com.ject.studytrip.mission.fixture.MissionFixture
import com.ject.studytrip.stamp.domain.model.Stamp
import org.springframework.stereotype.Component

@Component
class MissionTestHelper(
    private val missionRepository: MissionRepository,
) {
    fun saveMission(stamp: Stamp): Mission = missionRepository.save(MissionFixture(stamp).create())

    fun saveDeletedMission(stamp: Stamp): Mission = missionRepository.save(MissionFixture(stamp).create().also { it.updateDeletedAt() })

    fun saveCompletedMission(stamp: Stamp): Mission = missionRepository.save(MissionFixture(stamp).create().also { it.updateCompleted() })
}
